tests/e2e/utils/data_fetcher/traces.py

```diff
@@ -0,0 +1,266 @@
+    default_exclude_seg_name: List = ["Initialization", "Invocation", "Overhead"]
+
+    def __init__(
+        self,
+        filter_expression: str,
+        start_date: datetime,
+        end_date: Optional[datetime] = None,
+        xray_client: Optional[XRayClient] = None,
+        exclude_segment_name: Optional[List[str]] = None,
+        resource_name: Optional[List[str]] = None,
+        origin: Optional[List[str]] = None,
+        minimum_traces: int = 1,
+    ):
+        """Fetch and expose traces from X-Ray based on parameters
+
+        Data is recursively fetched in the following order:
+
+        * Trace summaries
+        * Trace IDs
+        * Traces
+        * Segments
+        * Subsegments
+        * Nested Subsegments
+
+        Parameters
+        ----------
+        filter_expression : str
+            AWS X-Ray Filter Expressions
+            see: https://docs.aws.amazon.com/xray/latest/devguide/xray-console-filters.html
+        start_date : datetime
+            Start date range to filter traces
+        end_date : Optional[datetime], optional
+            End date range to filter traces, by default 5 minutes past start_date
+        xray_client : Optional[XRayClient], optional
+            AWS X-Ray SDK Client, by default boto3.client('xray')
+        exclude_segment_name : Optional[List[str]], optional
+            Name of segments to exclude, by default ["Initialization", "Invocation", "Overhead"]
+        resource_name : Optional[List[str]], optional
+            Name of resource to filter traces (e.g., function name), by default None
+        origin : Optional[List[str]], optional
+            Trace origin name to filter traces, by default ["AWS::Lambda::Function"]
+        minimum_traces : int
+            Minimum number of traces to be retrieved before exhausting retry attempts
+        """
+        self.filter_expression = filter_expression
+        self.start_date = start_date
+        self.end_date = end_date or self.start_date + timedelta(minutes=5)
+        self.xray_client: XRayClient = xray_client or boto3.client("xray")
+        self.trace_ids: List[str] = []
+        self.trace_documents: Dict[str, TraceDocument] = {}
+        self.subsegments: List[TraceSubsegment] = []
+        self.exclude_segment_name = exclude_segment_name or self.default_exclude_seg_name
+        self.resource_name = resource_name
+        self.origin = origin or ["AWS::Lambda::Function"]
+        self.annotations: List[Dict[str, Any]] = []
+        self.metadata: List[Dict[str, Dict[str, Any]]] = []
+        self.minimum_traces = minimum_traces
+
+        paginator = self.xray_client.get_paginator("get_trace_summaries")
+        pages = paginator.paginate(
+            StartTime=self.start_date,
+            EndTime=self.end_date,
+            TimeRangeType="Event",
+            Sampling=False,
+            FilterExpression=self.filter_expression,
+        )
+
+        self.trace_ids = self._get_trace_ids(pages)
```

Comment:
I wonder why we store the result as class instance variable here? We use it only in one place as an input to get_trace_documents, so maybe passing this as a explicit parameter i.e self._get_trace_documents(trace_ids= self._get_trace_ids(pages)). What is your point of view on this?

Reply:
I was using in `test_tracer.py` too to quickly assert number of traces. Now that we ensure a minimum number of traces are found **and** that we assert on subsegments, we might not need it.

Removing it now. If we need later we can simply re-add it.